app/reducers/servers.js

```diff
@@ -273,6 +276,10 @@ function filterServers(filters, servers, state)
             filtered.push(server);
         }
     }
+    
+    if (state.favoriteServersOnly) {
+        filtered = filtered.filter((server) => Array.from(state.favoriteServers).find(e => e.guid === server.guid) != undefined);

```

Comment:
Sets have a `has` method that you can use to see if an item exists in them instead of iterating through them all. Also, I noticed an issue with my add / remove implementation. The `favoriteServers` state variable should hold server guids instead of entire server objects.

Reply:
Fixed this one.